### ISSUE TITLE
Nested Routes

### ISSUE THREAD

**Issue by bolthorn91** [jorgegorka/svelte-router#76]
Every time i try to navigate to a "nestedRoute" the navigation redirects to 404.html

My package.json has the --single flag to render. On my app the Main component has a <Route {currentRoute} {params} /> tag. I had provide the <Router/> tag and same problem is happening:

// Main.svelte

```
<script>
    import Header from '../components/Header.svelte'
    import { Route, Router } from 'svelte-router-spa'
    import { routes } from '../routes';

    export let currentRoute
    const params = {}
</script>

<Header/>

<Route {currentRoute}  {params} />
<!-- <Router/> -->
```

// package.json 
```
  "scripts": {
    "build": "rollup -c",
    "dev": "rollup -c -w",
    "start": "sirv public --single"
  },

```
// routes.ts
```
const routes = [
  {
    name: '/',
    component: Main,
    nestedRoutes: [
      { 
        name: 'budget/:id',
        component: Budget ,
      },
      { 
        name: 'cart',
        component: Cart ,
      }
    ],
  },
]
```

**Comment by jcschmidig** [comment]
Same here. It looks like nestedRoutes are not support at root level. Please document!

**Comment by beebase** [comment]
I'm also confused.
How can localhost:5000/page1 give me a 404 ?

![image](https://user-images.githubusercontent.com/446089/89908629-61ef6600-dbee-11ea-87c0-96a9240d2789.png)

**Comment by jorgegorka** [comment]
Hi @beebase 

The only unusual thing that I see is that Index is not a layout just a page so the route should be 

```:javascript
{ name: "page1", component: Page1 }
```

If you want to see a working example check this link:  https://github.com/jorgegorka/svelte-firebase/blob/master/src/lib/routes/public.js

Please not that layout components are different from page components.  In the former link you can see an example of both.

Cheers,

**Comment by beebase** [comment]
I think I might be missing something very obvious cause even with  a minimal setup, I still get 404 on localhost:5000/page1
Anyway thanks for the working example. Will look at that first.
![image](https://user-images.githubusercontent.com/446089/89937904-ff11c500-dc15-11ea-93c9-132a61ec868b.png)
![image](https://user-images.githubusercontent.com/446089/89938301-97a84500-dc16-11ea-8807-f413c815592f.png)

**Comment by jcschmidig** [comment]
Do you use snowpack? I found a strange error in that environment: replacing the folder name 'pages' with something else solved it.

**Comment by beebase** [comment]
Ok, got it all running. 
@jcschmidig  I'm using default Svelte/rollup.
@jorgegorka Thanks for creating this package. Coming from Vue, it feels very familiar.

Here are the mistakes I made being a newbie with Svelte
1)  package.json
do NOT forget to set the --single argument
"scripts": {
    "build": "rollup -c",
    "dev": "rollup -c -w",
    "start": "sirv public **--single**"
  },

2) when nesting pages, define the prop 'currentRoute' correctly
**export** let currentRoute = {}
![image](https://user-images.githubusercontent.com/446089/90004071-51de9180-dc95-11ea-8455-186430b76597.png)

😃 

![image](https://user-images.githubusercontent.com/446089/90004134-6e7ac980-dc95-11ea-84de-213394ff9b0e.png)

**Comment by beebase** [comment]
@bolthorn91 @jcschmidig @jorgegorka 
I think jcsschmidig is right, just tested  nestedRoutes in "/"  and that doesn't seem to work .

**Comment by SaltyDH** [comment]
Is anyone else having issues with sibling nested routes at three layers deep?

I added a couple of test routes to the svelte-firebase repo and whenever the navigation occurs, the address bar URL changes but the components do not swap.

I've tried both <a> and <Navigation> tags.

Example sibling routes:
/scenario/category/route1 : Component1
/scenario/category/route2 : Component2

The components are just a <div> with the component name in it.

Trying to work out if this is a known issue / limitation?
I'll try and get some time to submit more specifics but seemed to be potentially related.
